### ISSUE TITLE
[Enhancement]: Play a podcast library like a radio station

### ISSUE THREAD

**Issue by gordallott** [advplyr/audiobookshelf#1126]
### Describe the feature/enhancement

All I really want from my podcast apps is to be able to press play (usually by throwing on headphones which automatically does that), it resume whatever podcast I was previously playing and then once that is done, continue on to another podcast. 

I've solved this with other apps like PocketCasts in the past by enabling it's 'priority sort' feature for it's playlists. But it's buggy and weird and no one understands it. 

Really all I think most people want here is to be able to just say play my podcasts, pick the oldest/newest episode I haven't listened to yet (some people are backfilling, some people just want new stuff), and mix my podcasts up so that each time I get a different show. 

This is the main thing that is blocking me personally from picking up this as a backend, everything else seems great and pretty much perfect for my use

**Comment by advplyr** [comment]
I don't understand. A player queue was implemented on the web recently but not yet in the mobile app. Still trying to figure out the best way to implement. It looks like you are talking about a queue of episodes.

**Comment by advplyr** [comment]
As of v2.2.3 when you are playing a podcast episode you can add other episodes to the player queue.

**Comment by gordallott** [comment]
Yeah play queue is the basic first stepping stone most podcast apps take towards this feature. 

To explain better, to complete this in the somewhat awkward way that apps like podcast addict do (which isn't amazing but does work) that play queue would automatically be populated with episodes from all your podcasts, whilst juggling the series.

As an example this is my current playlist on Podcast Addict, I didn't add these episodes to the platlist, it does that automatically. Note also how there is never two episodes from the same series after each other
![Screenshot_20221113_143535_Podcast Addict](https://user-images.githubusercontent.com/2665595/201527331-75f6ed79-b0d0-4b16-a15f-829a3a232f4f.png)

**Comment by advplyr** [comment]
Just to avoid confusion, a Queue and a Playlist are going to be separate things in Abs.

I'm not sure what Podcast Addict is doing there but it looks like when new episodes are released they are added to the bottom of the queue.

**Comment by gordallott** [comment]
To be more specific, It's not quite that new episodes are added to the bottom, it's more that the queue is rebuilt when new episodes are added to play your podcasts from oldest to newest, whilst maintaining a mix of podcast sources. 

For example if you have three podcast sources, P1, P2, P3(all episodes listened to already) and your play queue looks like 

1. P1 2018/01/01
2. P2 2018/02/01
3. P1 2018/01/07
4. P2 2018/02/07

Then a new P3 episode is downloaded, then the play queue would look like this 


1. P1 2018/01/01
2. P2 2018/02/01
3. P3 2022/11/14
4. P1 2018/01/07
5. P2 2018/02/07

**Comment by advplyr** [comment]
Why would it be putting episodes in the queue that you listened to already?

**Comment by gordallott** [comment]
In the above example only the playlist source "P3" had episodes that had been listened to before, in the second example a new podcast was released in the "P3" source so it was added to the queue as you haven't listened to it yet

**Comment by advplyr** [comment]
I think this is the same as #1477 but is more clear to me. Let me know though

**Comment by advplyr** [comment]
Duplicate of #1477

**Comment by gordallott** [comment]
@advplyr (sorry for ping, but closed issues die, just wanted to update on if it's the same)

it's similar for sure, but without the ability to shuffle podcasts it wouldn't be the same feature. 

Sorting by date means that you'll only hear your oldest podcasts, whereas in Podcast Addict you can ask it to sort by shuffling your podcast sources - and then picking oldest episodes from each source. If there was a sort mode like that, it would be pretty much there :ok_hand:

**Comment by advplyr** [comment]
Shuffling and playing a podcast episode seems like a different feature request?

**Comment by gordallott** [comment]
So, in Podcast Addict this is called "Alternate Podcasts" and is found in the "Order by" section of your playlist. so you can order by various things like in the issue that duped this one.
But you can also specify that the order should take into account "Alternating your podcasts" so that you don't just get ten podcasts from podcast1 in a row just because that has more older episodes than podcast2 
![Screenshot_20230703_114414_Podcast Addict](https://github.com/advplyr/audiobookshelf/assets/2665595/69fa0763-0b32-46b2-acc5-6ee5a2a41fb1)

this may be a different feature request, I'm not sure! But the feature request this is marked as duped wouldn't fulfil this one without this capability
